packages/api-db-mongodb/src/db/user.ts

```diff
@@ -251,6 +259,78 @@ export class MongoDBUserAdapter implements DBUserAdapter {
       totalCount
     }
   }
+
+  async updateUserSubscription({
+    userId,
+    input
+  }: UpdateUserSubscriptionArgs): Promise<OptionalUserSubscription> {
+    const {
+      memberPlanId,
+      paymentPeriodicity,
+      monthlyAmount,
+      autoRenew,
+      startsAt,
+      payedUntil,
+      paymentMethod,
+      deactivatedAt
+    } = input
+    const {value} = await this.users.findOneAndUpdate(
+      {_id: userId},
+      {
+        $set: {
+          modifiedAt: new Date(),
+          'subscription.memberPlanId': memberPlanId,
+          'subscription.paymentPeriodicity': paymentPeriodicity,
+          'subscription.monthlyAmount': monthlyAmount,
+          'subscription.autoRenew': autoRenew,
+          'subscription.startsAt': startsAt,
+          'subscription.payedUntil': payedUntil,
+          'subscription.paymentMethod': paymentMethod,
+          'subscription.deactivatedAt': deactivatedAt
+        }
+      },
+      {returnOriginal: false}
+    )
+
+    return value?.subscription ? value.subscription : null
+  }
+
+  async deleteUserSubscription({userId}: DeleteUserSubscriptionArgs): Promise<string | null> {
+    const {value} = await this.users.findOneAndUpdate(
+      {_id: userId},
+      {
+        $set: {
+          modifiedAt: new Date()
+        },
+        $unset: {
+          subscription: ''
+        }
+      }
+    )
+
+    return value?._id
+  }
+
+  async updatePaymentProviderCustomers({
+    userID,
+    paymentProviderCustomers
+  }: UpdatePaymentProviderCustomerArgs): Promise<OptionalUser> {
+    const {value} = await this.users.findOneAndUpdate(
+      {_id: userID},
+      {
+        $set: {
+          modifiedAt: new Date(),
+          paymentProviderCustomers: paymentProviderCustomers
+        }
+      },
+      {returnOriginal: false}
+    )
+
+    if (!value) return null
+
+    const {_id: outID} = value
```

Comment:
What you see on line 331 and 332 is the shortest way of renaming the property of an object. `value` is an object with many properties one of them is `_id`. And i want all the properties from the object but `id` instead of `_id`. That's why i'm destructuring `_id` and rename the variable. Then i create a new object and put the value of `outID` to `id`. The reason i don't use directly id is because that variable is already used a bit further up.